docs/design/solutions/pce-weights-strategies/pce-weights-strategies.md

```diff
@@ -0,0 +1,19 @@
+# Refactor PCE with weights computation strategies
+
+## Goals
+For now PCE uses only cost field as graph edges weights
+
+While path computing logic going to be more complicated - we need to take several factors\costs to compute final edge\node weight, and keep this things maintainable.
+Also built AvailableNetwork should be logged, to make strange computed paths investigation possible.
+
+## Proposed changes
+- Introduce PCE data model instead of reusing DAO: for storing graph weighs by features and more convenient logging
+- Separate AvailableNetwork creation: it isn't responsibility of PathComputer
```

Comment:
I think that the PathComputer can have an interface where it either uses the default AvailableNetwork or is given an AvailableNetwork. I believe the code is structured like that already. The benefit of using a default is that it doesn't require every PC client to call the same method to get available network; IOW, it makes sense to provide a simplified interface, and provide a more complex one for those that need it.

But, why do this as part of PCE diversity? What's the relationship, other than you're working on this code? It could be better to lodge this as a separate improvement proposal, to be discussed independent of diversity.

Reply:
Ok, now I am also decline to keep single PCE interface and do all work inside.

Diversity is here, because this design was introduced in debates about how it should be implemented, and it going to be the first feature what will use it.
Aim is to have one computer class what will use strategies to calculate graph edges "weights" at runtime. The simplest strategy is "ISL cost getter", what is equal with the current PCE implementation.